apps/prairielearn/src/sync/fromDisk/authors.ts

```diff
@@ -0,0 +1,62 @@
+import * as sqldb from '@prairielearn/postgres';
+import { callAsync } from '@prairielearn/postgres';
+
+import { type Author, AuthorSchema } from '../../lib/db-types.js';
+import { type CourseData } from '../course-db.js';
+import * as infofile from '../infofile.js';
+
+const sql = sqldb.loadSqlEquiv(import.meta.url);
+
+export async function sync(
+  courseId: string,
+  courseData: CourseData,
+  questionIds: Record<string, string>,
+) {
+  if (infofile.hasErrors(courseData.course)) {
+    return;
+  }
+
+  // Collect all unique authors from all questions
+  const uniqueAuthors = new Set<string>();
+  for (const qid of Object.keys(courseData.questions)) {
+    const question = courseData.questions[qid];
+    if (infofile.hasErrors(question)) continue;
+    const authors = question.data?.authors ?? [];
+    for (const author of authors) {
+      uniqueAuthors.add(author);
+    }
+  }
+
+  // Insert all unique authors
+  await sqldb.queryAsync(sql.insert_authors, {
+    authors: Array.from(uniqueAuthors).map((authorString) => JSON.stringify([authorString])),
+  });
+
```

Comment:
Do I need to use the new `determineOperationsForEntities` that is being used for tags and topics? It seemed like overkill since authors won't be updated or removed, only added.